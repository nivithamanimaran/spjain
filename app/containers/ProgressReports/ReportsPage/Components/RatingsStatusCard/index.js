import React from 'react';
import { Grid, makeStyles, Box, useTheme, Typography } from '@material-ui/core';
import ArrowUpwardIcon from '@material-ui/icons/ArrowUpward';
import ArrowDownwardIcon from '@material-ui/icons/ArrowDownward';
import RemoveCircleIcon from '@material-ui/icons/RemoveCircle';
import ArrowForwardIcon from '@material-ui/icons/ArrowForward';
import { useTranslation } from 'react-i18next';

const useStyles = makeStyles(theme => ({
  cardContainer: {
    display: 'flex',
    justifyContent: 'space-between',
  },
  cardTitle: {
    color: theme.palette.KenColors.tertiaryYellow501,
    fontWeight: 'normal',
    fontSize: '12px',
    lineHeight: '20px',
  },
  cardAction: {
    color: theme.palette.KenColors.tertiaryYellow501,
    fontWeight: 'bold',
    cursor: 'pointer',
    height: '15px',
    width: '15px',
  },
  cardContent: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'flex-end',
  },
  percentage: {
    fontWeight: 'bold',
    color: theme.palette.KenColors.tertiaryYellow502,
  },
  percentageSymbol: {
    fontWeight: 'bold',
    color: theme.palette.KenColors.tertiaryYellow502,
  },
  ratingsContainer: {
    display: 'flex',
    alignItems: 'center',
    padding: '5px',
    color: theme.palette.KenColors.tertiaryGreen503,
  },
  upArrow: {
    backgroundColor: theme.palette.KenColors.tertiaryGreen504,
    borderRadius: '50%',
    color: theme.palette.KenColors.kenWhite,
    height: '15px',
    width: '15px',
  },
  downArrow: {
    backgroundColor: theme.palette.KenColors.secondaryPeach2,
    borderRadius: '50%',
    color: theme.palette.KenColors.tertiaryRed501,
    height: '15px',
    width: '15px',
  },
  stableArrow: {
    backgroundColor: theme.palette.KenColors.tertiaryBlue100,
    borderRadius: '50%',
    color: theme.palette.KenColors.kenWhite,
    height: '15px',
    width: '15px',
  },
  forwardArrow: {
    backgroundColor: theme.palette.KenColors.tertiaryBlue100,
    borderRadius: '50%',
    color: theme.palette.KenColors.kenWhite,
    height: '15px',
    width: '15px',
  },
  ratings: {
    fontWeight: 600,
    paddingLeft: '5px',
  },
}));

const RatingsStatusCard = props => {
  const {
    cardBackgroundColor,
    title,
    titleColor,
    cardAction,
    children,
    boldText,
    boldSubtext,
    status, //increased, decreased, stable
    ratings,
    ratingsText,
    ratingsSymbol,
    onIconClick,
    ratingsIcon,
    variant,
    ratingsTextColor,
    ratingsStatusTextColor,
  } = props;
  const { t } = useTranslation();
  const theme = useTheme();
  const classes = useStyles();

  const getStylesByStatus = key => {
    let obj;
    switch (status) {
      case 'stable': {
        obj = {
          icon: (
            <RemoveCircleIcon
              fontSize="small"
              className={classes.stableArrow}
            />
          ),
          ratingsText: t('labels:Rating_Stable'),
          ratingsStatusTextColor: theme.palette.KenColors.tertiaryBlue100,
          titleColor: theme.palette.KenColors.tertiaryYellow502,
        };
        break;
      }
      case 'increased': {
        obj = {
          icon: (
            <ArrowUpwardIcon fontSize="small" className={classes.upArrow} />
          ),
          ratingsText: t('labels:Rating_Up'),
          ratingsStatusTextColor: theme.palette.KenColors.tertiaryBlue100,
          titleColor: theme.palette.KenColors.tertiaryYellow502,
        };
        break;
      }
      case 'decreased': {
        obj = {
          icon: (
            <ArrowDownwardIcon fontSize="small" className={classes.downArrow} />
          ),
          ratingsText: t('labels:Rating_Down'),
          ratingsStatusTextColor: theme.palette.KenColors.tertiaryBlue100,
          titleColor: theme.palette.KenColors.tertiaryYellow502,
        };
        break;
      }

      default: {
        obj = {
          icon: null,
          ratingsText: t('labels:Rating_Default'),
          ratingsStatusTextColor: theme.palette.KenColors.tertiaryBlue100,
          titleColor: theme.palette.KenColors.tertiaryYellow502,
        };
        break;
      }
    }

    if (key === 'ratingsText' && ratingsText) return ratingsText;
    if (key === 'ratingsIcon' && ratingsIcon) return ratingsIcon;
    if (key === 'ratingsStatusTextColor' && ratingsStatusTextColor)
      return ratingsStatusTextColor;
    return obj[key];
  };

  const getStylesByVariant = key => {
    let obj;
    switch (variant) {
      case 'primary': {
        obj = {
          backgroundColor: theme.palette.KenColors.tertiaryYellow74,
          titleColor: theme.palette.KenColors.tertiaryYellow502,
          ratingsTextColor: theme.palette.KenColors.tertiaryYellow502,
        };
        break;
      }
      case 'secondary': {
        obj = {
          backgroundColor: theme.palette.KenColors.tertiaryTeal198,
          titleColor: theme.palette.KenColors.tertiaryTeal501,
          ratingsTextColor: theme.palette.KenColors.tertiaryTeal501,
        };
        break;
      }
      case 'tertiary': {
        obj = {
          backgroundColor: theme.palette.KenColors.tertiaryPurple70,
          titleColor: theme.palette.KenColors.tertiaryPurple501,
          ratingsTextColor: theme.palette.KenColors.tertiaryPurple501,
        };
        break;
      }
      case 'quaternary': {
        obj = {
          backgroundColor: theme.palette.KenColors.secondaryPeach5,
          titleColor: theme.palette.KenColors.secondaryPeach6,
          ratingsTextColor: theme.palette.KenColors.secondaryPeach6,
        };
        break;
      }
      case 'quinary': {
        obj = {
          backgroundColor: theme.palette.KenColors.tertiaryGreen49,
          titleColor: theme.palette.KenColors.tertiaryGreen48,
          ratingsTextColor: theme.palette.KenColors.tertiaryGreen48,
        };
        break;
      }

      default: {
        obj = {
          backgroundColor: theme.palette.KenColors.kenWhite,
          titleColor: theme.palette.KenColors.tertiaryBlue501,
          ratingsTextColor: theme.palette.KenColors.tertiaryBlue501,
        };
        break;
      }
    }

    if (key === 'backgroundColor' && cardBackgroundColor)
      return cardBackgroundColor;
    if (key === 'titleColor' && titleColor) return titleColor;
    if (key === 'ratingsTextColor' && ratingsTextColor) return ratingsTextColor;
    return obj[key];
  };

  return (
    <Box
      style={{
        borderRadius: '10px',
        backgroundColor: getStylesByVariant('backgroundColor'),
      }}
      data-testid="ratings-status-card"
    >
      <Grid container spacing={0}>
        <Grid item xs={12}>
          <Box p={1} pl={2} className={classes.cardContainer}>
            <Typography
              data-testid="card-title"
              className={classes.cardTitle}
              variant="subtitle2"
              style={{ color: getStylesByVariant('titleColor') }}
            >
              {title}
            </Typography>
            {cardAction || (
              <ArrowForwardIcon
                className={classes.cardAction}
                fontSize="small"
                onClick={onIconClick}
                style={{ color: getStylesByVariant('titleColor') }}
              />
            )}
          </Box>
        </Grid>
        {(boldText || boldSubtext || ratings || ratingsSymbol) && (
          <Grid item xs={12}>
            <Box p={1} pl={2} className={classes.cardContent}>
              <Grid container spacing={1}>
                {(boldText || boldSubtext) && (
                  <Grid item xs={6}>
                    <Typography
                      data-testid="card-boldText"
                      className={classes.percentage}
                      variant="h4"
                      component="span"
                      style={{ color: getStylesByVariant('ratingsTextColor') }}
                    >
                      {boldText}
                    </Typography>
                    <Typography
                      data-testid="card-boldSubtext"
                      className={classes.percentageSymbol}
                      variant="h6"
                      component="span"
                      style={{ color: getStylesByVariant('ratingsTextColor') }}
                    >
                      {boldSubtext}
                    </Typography>
                  </Grid>
                )}
                {(ratings || ratingsSymbol) && (
                  <Grid item xs={6} className={classes.ratingsContainer}>
                    {getStylesByStatus('icon')}
                    <Typography
                      className={classes.ratings}
                      variant="subtitle2"
                      style={{
                        color: getStylesByStatus('ratingsStatusTextColor'),
                      }}
                    >
                      {getStylesByStatus('ratingsText')} {ratings}
                      {ratingsSymbol || t('labels:Rating_Symbol_Default')}
                    </Typography>
                  </Grid>
                )}
              </Grid>
            </Box>
          </Grid>
        )}
        {children && (
          <Grid item xs={12}>
            {children}
          </Grid>
        )}
      </Grid>
    </Box>
  );
};
export default RatingsStatusCard;
