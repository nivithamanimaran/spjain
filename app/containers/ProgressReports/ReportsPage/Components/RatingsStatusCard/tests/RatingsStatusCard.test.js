import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import ReactDOM from 'react-dom';
import RatingsStatusCard from '../index';
import ThemeProvider from '../../../../../../utils/themeProvider';
import { BrowserRouter as Router } from 'react-router-dom';

afterEach(() => {
  cleanup();
});

describe('Ratings-Status-Card', () => {
  const div = document.createElement('div');
  it('Ratings Status Card renders correctly without props', () => {
    ReactDOM.render(
      <ThemeProvider>
        <Router>
          <RatingsStatusCard />
        </Router>
      </ThemeProvider>,
      div
    );

    const { getByTestId } = render(
      <ThemeProvider>
        <Router>
          <RatingsStatusCard />
        </Router>
      </ThemeProvider>
    );

    expect(getByTestId('ratings-status-card')).toMatchSnapshot();
  });
  it('Ratings status Card renders correctly with props', () => {
    ReactDOM.render(
      <ThemeProvider>
        <Router>
          <RatingsStatusCard boldText="bold" boldSubtext="subtext" />
        </Router>
      </ThemeProvider>,
      div
    );

    const { queryByTestId } = render(
      <ThemeProvider>
        <Router>
          <RatingsStatusCard
            title="Title"
            boldText="bold"
            boldSubtext="subtext"
          />
        </Router>
      </ThemeProvider>
    );

    const title = queryByTestId('card-title');
    expect(title).toBeTruthy();
    expect(title.textContent).toBe('Title');

    const bold = queryByTestId('card-boldText');
    expect(bold).toBeTruthy();
    expect(bold.textContent).toBe('bold');

    const subText = queryByTestId('card-boldSubtext');
    expect(subText).toBeTruthy();
    expect(subText.textContent).toBe('subtext');
  });
});
