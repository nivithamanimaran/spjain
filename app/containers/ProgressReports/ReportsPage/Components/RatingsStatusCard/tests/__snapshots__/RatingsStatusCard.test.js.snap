// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ratings-Status-Card Ratings Status Card renders correctly 1`] = `
<div
  class="MuiBox-root MuiBox-root-16"
  data-testid="ratings-status-card"
  style="border-radius: 10px; background-color: rgb(230, 252, 255);"
>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiBox-root MuiBox-root-17 makeStyles-cardContainer-1"
      >
        <h6
          class="MuiTypography-root makeStyles-cardTitle-2 MuiTypography-subtitle2"
          style="color: rgb(0, 33, 141);"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-cardAction-3 MuiSvgIcon-fontSizeSmall"
          focusable="false"
          style="color: rgb(0, 33, 141);"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
          />
        </svg>
      </div>
      <div
        class="MuiBox-root MuiBox-root-18 makeStyles-cardContent-4"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
          >
            <span
              class="MuiTypography-root makeStyles-percentage-5 MuiTypography-h4"
              style="color: rgb(0, 33, 141);"
            />
            <span
              class="MuiTypography-root makeStyles-percentageSymbol-6 MuiTypography-h6"
              style="color: rgb(0, 33, 141);"
            />
          </div>
          <div
            class="MuiGrid-root makeStyles-ratingsContainer-7 MuiGrid-item MuiGrid-grid-xs-6"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-forwardArrow-11 MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
              />
            </svg>
            <h6
              class="MuiTypography-root makeStyles-ratings-12 MuiTypography-subtitle2"
              style="color: rgb(0, 33, 141);"
            >
               
              labels:Rating_Symbol_Default
            </h6>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Ratings-Status-Card Ratings Status Card renders correctly without props 1`] = `
<div
  class="MuiBox-root MuiBox-root-16"
  data-testid="ratings-status-card"
  style="border-radius: 10px; background-color: rgb(230, 252, 255);"
>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiBox-root MuiBox-root-17 makeStyles-cardContainer-1"
      >
        <h6
          class="MuiTypography-root makeStyles-cardTitle-2 MuiTypography-subtitle2"
          data-testid="card-title"
          style="color: rgb(0, 33, 141);"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-cardAction-3 MuiSvgIcon-fontSizeSmall"
          focusable="false"
          style="color: rgb(0, 33, 141);"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
          />
        </svg>
      </div>
      <div
        class="MuiBox-root MuiBox-root-18 makeStyles-cardContent-4"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
          >
            <span
              class="MuiTypography-root makeStyles-percentage-5 MuiTypography-h4"
              data-testid="card-boldText"
              style="color: rgb(0, 33, 141);"
            />
            <span
              class="MuiTypography-root makeStyles-percentageSymbol-6 MuiTypography-h6"
              data-testid="card-boldSubtext"
              style="color: rgb(0, 33, 141);"
            />
          </div>
          <div
            class="MuiGrid-root makeStyles-ratingsContainer-7 MuiGrid-item MuiGrid-grid-xs-6"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-forwardArrow-11 MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
              />
            </svg>
            <h6
              class="MuiTypography-root makeStyles-ratings-12 MuiTypography-subtitle2"
              style="color: rgb(0, 33, 141);"
            >
               
              labels:Rating_Symbol_Default
            </h6>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
